File: 프로그래머스/1/42840. 모의고사/모의고사.java
import java.util.*;
class Solution {
    int one[] = {1,2,3,4,5};
    int two[] = {2,1,2,3,2,4,2,5};
    int three[] = {3,3,1,1,2,2,4,4,5,5};
    
    public int[] solution(int[] answers) {
        int[] answer;
        
        int maxScore = 0;
        int[] scores = new int[4];
        int idx1 = 0; int idx2 = 0; int idx3 = 0;
        for(int i=0; i<answers.length; i++) {
            int ans = answers[i];
            if(ans == one[idx1]) scores[1]++;
            if(ans == two[idx2]) scores[2]++;
            if(ans == three[idx3]) scores[3]++;
            maxScore = Math.max(scores[1], Math.max(scores[2], scores[3]));
            
            idx1 = (idx1+1)%5;
            idx2 = (idx2+1)%8;
            idx3 = (idx3+1)%10;
        }
        
        ArrayList<Integer> arr = new ArrayList<>();
        for(int i=1; i<=3; i++) {
            if(scores[i] == maxScore) arr.add(i);
        }
        
        answer = new int[arr.size()];
        for(int i=0; i<arr.size(); i++) {
            answer[i] = arr.get(i);
        }
        
        return answer;
    }
}

File: 프로그래머스/1/42840. 모의고사/모의고사.cpp
#include <string>
#include <vector>

using namespace std;

int nums1[] = {1,2,3,4,5};
int nums2[] = {1,3,4,5};
int nums3[] = {3,1,2,4,5};

vector<int> solution(vector<int> answers) {
    vector<int> answer;
    int size = answers.size();
    vector<int> one(size,0); vector<int> two(size,0); vector<int> three(size,0);
    vector<int> scores(4,0);
    
    // 1.
    int d = 0;
    for(int i=0; i<size; i++) {
        one[i] = nums1[d];
        d = (d+1)%5;
    }
    
    // 2.
    for(int i=0; i<size; i+=2) {
        two[i] = 2;
    }
    d = 0;
    for(int i=1; i<size; i+=2) {
        two[i] = nums2[d];
        d = (d+1)%4;
    }
    
    // 3. 
    d = 0;
    for(int i=0; i<size; i+=2) {
        three[i] = nums3[d]; 
        if(i+1 < size) three[i+1] = nums3[d];
        d = (d+1)%5;
    }
    
    
    for(int i=0; i<size; i++) {
        if(one[i] == answers[i]) scores[1]++;
        if(two[i] == answers[i]) scores[2]++;
        if(three[i] == answers[i]) scores[3]++;
    }
    int maxScore = max(scores[1], max(scores[2], scores[3]));
    
    for(int i=1; i<=3; i++) {
        if(scores[i] == maxScore) answer.push_back(i);
    }
    
    return answer;
}